src/System.Runtime/tests/System/Reflection/TypeDelegatorTests.netcoreapp.cs

```diff
@@ -1,40 +1,113 @@
 using System.Collections.Generic;
+using System.Linq;
 using Xunit;
 
 namespace System.Reflection.Tests
 {
     public class TypeDelegatorNetcoreTests
     {
-        private static IEnumerable<object[]> SZArrayOrNotTypes()
-        {
-            yield return new object[] { typeof(int[]), true };
-            yield return new object[] { typeof(string[]), true };
-            yield return new object[] { typeof(void), false };
-            yield return new object[] { typeof(int), false };
-            yield return new object[] { typeof(int[]).MakeByRefType(), false };
-            yield return new object[] { typeof(int[,]), false };
-            yield return new object[] { typeof(TypeInfoTests), false };
-            yield return new object[] { Array.CreateInstance(typeof(int), new[] { 2 }, new[] { -1 }).GetType(), false };
-            yield return new object[] { Array.CreateInstance(typeof(int), new[] { 2 }, new[] { 1 }).GetType(), false };
-            yield return new object[] { Array.CreateInstance(typeof(int), new[] { 2 }, new[] { 0 }).GetType(), true };
-            yield return new object[] { typeof(int[][]), true };
-            yield return new object[] { Type.GetType("System.Int32[]"), true };
-            yield return new object[] { Type.GetType("System.Int32[*]"), false };
-            yield return new object[] { Type.GetType("System.Int32"), false };
-            yield return new object[] { typeof(int).MakeArrayType(), true };
-            yield return new object[] { typeof(int).MakeArrayType(1), false };
-            yield return new object[] { typeof(int).MakeArrayType().MakeArrayType(), true };
-            yield return new object[] { typeof(int).MakeArrayType(2), false };
-            yield return new object[] { typeof(Outside<int>.Inside<string>), false };
-            yield return new object[] { typeof(Outside<int>.Inside<string>[]), true };
-            yield return new object[] { typeof(Outside<int>.Inside<string>[,]), false };
-            yield return new object[] { Array.CreateInstance(typeof(Outside<int>.Inside<string>), new[] { 2 }, new[] { -1 }).GetType(), false };
-        }
-
-        [Theory, MemberData(nameof(SZArrayOrNotTypes))]
-        public void IsSZArray(Type type, bool expected)
-        {
-            Assert.Equal(expected, new TypeDelegator(type).IsSZArray);
+        private static readonly IEnumerable<Type> NonArrayBaseTypes = new Type[]
+         {
+            typeof(int),
+            typeof(void),
+            typeof(int*),
+            typeof(Outside),
+            typeof(Outside<>),
+            typeof(Outside<>).GetTypeInfo().GenericTypeParameters[0],
+            Type.GetTypeFromCLSID(default(Guid)),
+            new object().GetType().GetType()
+         };
+
+        [Fact]
+        public void IsSZArray_FalseForNonArrayTypes()
+        {
+            foreach (Type type in NonArrayBaseTypes)
+            {
+                Assert.False(new TypeDelegator(type).IsSZArray);
```

Comment:
OK, in the interests of getting this finished, I'll leave the TypeDelegator tests as they were for now.  Is it worth tracking in an issue, or will it be picked up when the Type stuff gets cleaned up?